common-lib/src/main/scala/com/gu/mediaservice/lib/aws/Bedrock.scala

```diff
@@ -0,0 +1,92 @@
+package com.gu.mediaservice.lib.aws
+
+import software.amazon.awssdk.services.bedrockruntime.model._
+import software.amazon.awssdk.services.bedrockruntime._
+import com.gu.mediaservice.lib.config.CommonConfig
+import play.api.libs.json.Json
+import software.amazon.awssdk.core.SdkBytes
+
+import java.net.URI
+import com.gu.mediaservice.lib.logging.LogMarker
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.OFormat.oFormatFromReadsAndOWrites
+import play.api.libs.json._
+
+object Bedrock {
+  private case class BedrockRequest(
+   input_type: String,
+   embedding_types: List[String],
+   images: List[String]
+ )
+  private implicit val bedrockRequestFormat: OFormat[BedrockRequest] = Json.format[BedrockRequest]
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.FutureConverters.CompletionStageOps
+
+class Bedrock(config: CommonConfig)
+  extends AwsClientV2BuilderUtils {
+
+  // TODO: figure out what the more usual pattern for turning off localstack behaviour is
+  override def awsLocalEndpointUri: Option[URI] = None
+
+  override def isDev: Boolean = config.isDev
+
+  val client: BedrockRuntimeClient = {
+    withAWSCredentialsV2(BedrockRuntimeClient.builder())
+      .build()
+  }
+
+  private def createRequestBody(base64EncodedImage: String): InvokeModelRequest = {
+    val body = Bedrock.BedrockRequest(
+      input_type = "image",
+      embedding_types = List("float"),
+      images = List(s"data:image/jpg;base64,$base64EncodedImage")
```

Comment:
Ah! Can we be certain that every image is a JPEG? I think the Grid does technically support PNGs, though I feel like they are rare...

Reply:
well remembered! no you can't, we'll see both .png and .tiff files

It looks like bedrock/cohere won't accept .tiff files so we shouldn't send those. Fortunately for other reasons during the pipeline we'll create a jpg/png "browser-viewable" version of a tiff, so we can send that instead. That'll require a slightly different wiring in Uploader/Projector). We can look at that next weds?



(Also it looks like they only accept images up to 5MB, so we'll need to deal with that too)

Reply:
Just to play "ruthless MVP's advocate" for a sec, what if initially we could just filter out images that are not JPEGs and over 5MB? (Depending on how big a proportion of incoming images these constitute)

This is obviously a terrible idea for the real production feature, but until we do a backfill we're going to have swathes of unembedded images anyway so the prototype feature will be: "of what we've embedded so far, here are the best 30". Given that, maaaaybe it's OK to kick this down the road a little, especially because we may shortly be upgrading to [Cohere v4](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed-v4.html) and having to re-embed everything